lib/internal/encoding.js

```diff
@@ -0,0 +1,429 @@
+  ['windows-1251', 'windows-1251'],
+  ['x-cp1251', 'windows-1251'],
+  ['ansi_x3.4-1968', 'windows-1252'],
+  ['ascii', 'windows-1252'],
+  ['cp1252', 'windows-1252'],
+  ['cp819', 'windows-1252'],
+  ['csisolatin1', 'windows-1252'],
+  ['ibm819', 'windows-1252'],
+  ['iso-8859-1', 'windows-1252'],
+  ['iso-ir-100', 'windows-1252'],
+  ['iso8859-1', 'windows-1252'],
+  ['iso88591', 'windows-1252'],
+  ['iso_8859-1', 'windows-1252'],
+  ['iso_8859-1:1987', 'windows-1252'],
+  ['l1', 'windows-1252'],
+  ['latin1', 'windows-1252'],
+  ['us-ascii', 'windows-1252'],
+  ['windows-1252', 'windows-1252'],
+  ['x-cp1252', 'windows-1252'],
+  ['cp1253', 'windows-1253'],
+  ['windows-1253', 'windows-1253'],
+  ['x-cp1253', 'windows-1253'],
+  ['cp1254', 'windows-1254'],
+  ['csisolatin5', 'windows-1254'],
+  ['iso-8859-9', 'windows-1254'],
+  ['iso-ir-148', 'windows-1254'],
+  ['iso8859-9', 'windows-1254'],
+  ['iso88599', 'windows-1254'],
+  ['iso_8859-9', 'windows-1254'],
+  ['iso_8859-9:1989', 'windows-1254'],
+  ['l5', 'windows-1254'],
+  ['latin5', 'windows-1254'],
+  ['windows-1254', 'windows-1254'],
+  ['x-cp1254', 'windows-1254'],
+  ['cp1255', 'windows-1255'],
+  ['windows-1255', 'windows-1255'],
+  ['x-cp1255', 'windows-1255'],
+  ['cp1256', 'windows-1256'],
+  ['windows-1256', 'windows-1256'],
+  ['x-cp1256', 'windows-1256'],
+  ['cp1257', 'windows-1257'],
+  ['windows-1257', 'windows-1257'],
+  ['x-cp1257', 'windows-1257'],
+  ['cp1258', 'windows-1258'],
+  ['windows-1258', 'windows-1258'],
+  ['x-cp1258', 'windows-1258'],
+  ['x-mac-cyrillic', 'x-mac-cyrillic'],
+  ['x-mac-ukrainian', 'x-mac-cyrillic'],
+  ['chinese', 'gbk'],
+  ['csgb2312', 'gbk'],
+  ['csiso58gb231280', 'gbk'],
+  ['gb2312', 'gbk'],
+  ['gb_2312', 'gbk'],
+  ['gb_2312-80', 'gbk'],
+  ['gbk', 'gbk'],
+  ['iso-ir-58', 'gbk'],
+  ['x-gbk', 'gbk'],
+  ['gb18030', 'gb18030'],
+  ['big5', 'big5'],
+  ['big5-hkscs', 'big5'],
+  ['cn-big5', 'big5'],
+  ['csbig5', 'big5'],
+  ['x-x-big5', 'big5'],
+  ['cseucpkdfmtjapanese', 'euc-jp'],
+  ['euc-jp', 'euc-jp'],
+  ['x-euc-jp', 'euc-jp'],
+  ['csiso2022jp', 'iso-2022-jp'],
+  ['iso-2022-jp', 'iso-2022-jp'],
+  ['csshiftjis', 'shift_jis'],
+  ['ms932', 'shift_jis'],
+  ['ms_kanji', 'shift_jis'],
+  ['shift-jis', 'shift_jis'],
+  ['shift_jis', 'shift_jis'],
+  ['sjis', 'shift_jis'],
+  ['windows-31j', 'shift_jis'],
+  ['x-sjis', 'shift_jis'],
+  ['cseuckr', 'euc-kr'],
+  ['csksc56011987', 'euc-kr'],
+  ['euc-kr', 'euc-kr'],
+  ['iso-ir-149', 'euc-kr'],
+  ['korean', 'euc-kr'],
+  ['ks_c_5601-1987', 'euc-kr'],
+  ['ks_c_5601-1989', 'euc-kr'],
+  ['ksc5601', 'euc-kr'],
+  ['ksc_5601', 'euc-kr'],
+  ['windows-949', 'euc-kr'],
+  ['utf-16be', 'utf-16be'],
+  ['utf-16le', 'utf-16le'],
+  ['utf-16', 'utf-16le'],
+  // Unsupported standard mappings
+  // ['csiso2022kr', 'replacement'],
+  // ['hz-gb-2312', 'replacement'],
+  // ['iso-2022-cn', 'replacement'],
+  // ['iso-2022-cn-ext', 'replacement'],
+  // ['iso-2022-kr', 'replacement'],
```

Comment:
Rather than mentioning these as "unsupported", we can in fact say they don't necessarily need to be supported, since `TextDecoder` errors out on `'replacement'`.